script/prerender/createRedirects.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import path from 'path';
+import { Redirects } from '../../data/redirects/types';
+import { content } from '../../src/app/content/routes';
+import { makeUnifiedBookLoader } from '../../src/app/content/utils';
+import { findArchiveTreeNodeById } from '../../src/app/content/utils/archiveTreeUtils';
+import { AppServices } from '../../src/app/types';
+import config from '../../src/config.books';
+import { writeAssetFile } from './fileUtils';
+
+const redirectsPath = path.resolve(__dirname, '../../data/redirects/');
+
+const createRedirects = async(archiveLoader: AppServices['archiveLoader'], osWebLoader: AppServices['osWebLoader']) => {
+  const bookLoader = makeUnifiedBookLoader(archiveLoader, osWebLoader);
+
+  const books = fs.readdirSync(redirectsPath).filter((name) => name.match('.json'));
+
+  const redirects: Array<{ from: string, to: string }> = [];
+
+  for (const fileName of books) {
+    const bookRedirects: Redirects = await import(`${redirectsPath}/${fileName}`);
+    const bookIdFromFilename = fileName.replace('.json', '');
+    const configForBook: { defaultVersion: string } | undefined = config[bookIdFromFilename];
+
+    if (!configForBook) {
+      // tslint:disable-next-line: no-console
+      console.log(`Couldn't find version for book: ${bookIdFromFilename}`);
+      continue;
+    }
+
+    const { tree, slug: bookSlug } = await bookLoader(bookIdFromFilename, configForBook.defaultVersion);
```

Comment:
yea my intent was to not use the filename for anything. i see now that would mean moving the async call into the page loop.... i guess thats probably fine because of the caching. i like that it removes the assumption that the files are broken down by book and uses the bookId that was put in the json structure